output/transform/util.go

```diff
@@ -61,8 +62,16 @@ func upsertQueryReferenceAndInformation(s *snapshot.FullSnapshot, statementTexts
 	return idx
 }
 
-func upsertQueryReferenceAndInformationSimple(server *state.Server, refs []*snapshot.QueryReference, infos []*snapshot.QueryInformation, roleIdx int32, databaseIdx int32, originalQuery string, trackActivityQuerySize int) (int32, []*snapshot.QueryReference, []*snapshot.QueryInformation) {
-	fingerprint := util.FingerprintQuery(originalQuery, server.Config.FilterQueryText, trackActivityQuerySize)
+func upsertQueryReferenceAndInformationSimple(server *state.Server, refs []*snapshot.QueryReference, infos []*snapshot.QueryInformation, roleIdx int32, databaseIdx int32, queryID null.Int, originalQuery string, trackActivityQuerySize int) (int32, []*snapshot.QueryReference, []*snapshot.QueryInformation) {
+	var fingerprint uint64
+	if server.PrevState.QueryIdentities != nil && queryID.Valid {
+		if identity, ok := server.PrevState.QueryIdentities[queryID.Int64]; ok {
+			fingerprint = identity.Fingerprint
+		}
+	}
+	if fingerprint == 0 {
```

Comment:
Are you sure `0` isn't a valid hash value?

Reply:
How would I confirm that? `PgQuery.fingerprint('')` in Ruby generates `d8d13f8b2da6c9ad`

Reply:
I think thats really a [XXH3](https://github.com/Cyan4973/xxHash) question - and since I couldn't find any particular documentation that states `0` is not a possible digest result, I think we should assume it could be.

Reply:
Even if it's theoretically possible, surely it wouldn't be a valid query?

But let's say it's a needed change. How would we have a null type here when numbers in Go default to zero? Use a `NullInt`?

Reply:
> Even if it's theoretically possible, surely it wouldn't be a valid query?

Why couldn't it be a valid query? I don't know the necessary tokens that'd need to be the input to the fingerprint function, but it's possible that the right combination of token values produces a hash value of `0` - it's just as likely as producing e.g. a value of `42` or `123`.

> But let's say it's a needed change. How would we have a null type here when numbers in Go default to zero? Use a NullInt? Add a local boolean variable to check if it's already been set?

I think in this particular case, if we just need to know inside the function, we could just use a local boolean variable that indicates whether we found any fingerprint.